src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,337 @@
+                            return true;
+                        }
+                    }
+
+                    return false;
+                }
+
+                ImmutableHashSet<ISymbol> GetCandidateSymbolsReferencedInDocComments()
+                {
+                    var builder = ImmutableHashSet.CreateBuilder<ISymbol>();
+                    foreach (var root in symbolEndContext.Symbol.Locations.Select(l => l.SourceTree.GetRoot(symbolEndContext.CancellationToken)))
+                    {
+                        SemanticModel lazyModel = null;
+                        foreach (var node in root.DescendantNodes(descendIntoTrivia: true)
+                                                 .OfType<TDocumentationCommentTriviaSyntax>()
+                                                 .SelectMany(n => n.DescendantNodes().OfType<TIdentifierNameSyntax>()))
+                        {
+                            lazyModel = lazyModel ?? symbolEndContext.Compilation.GetSemanticModel(root.SyntaxTree);
+                            var symbol = lazyModel.GetSymbolInfo(node, symbolEndContext.CancellationToken).Symbol;
+                            if (symbol != null && IsCandidateSymbol(symbol))
+                            {
+                                builder.Add(symbol);
+                            }
+                        }
+                    }
+
+                    return builder.ToImmutable();
+                }
+            }
+
+            private bool IsCandidateSymbol(ISymbol memberSymbol)
+            {
+                if (memberSymbol.DeclaredAccessibility == Accessibility.Private &&
+                    !memberSymbol.IsImplicitlyDeclared)
+                {
+                    // Do not track accessors, as we will track the associated symbol.
+                    // Also skip entry point (Main) method
+                    switch (memberSymbol.Kind)
+                    {
+                        case SymbolKind.Method:
+                            var methodSymbol = (IMethodSymbol)memberSymbol;
+                            return methodSymbol.AssociatedSymbol == null && !IsEntryPoint(methodSymbol);
+
+                        case SymbolKind.Field:
+                            return ((IFieldSymbol)memberSymbol).AssociatedSymbol == null;
+
+                        default:
+                            return true;
+                    }
+                }
+
+                return false;
+
+                // Local functions.
+                bool IsEntryPoint(IMethodSymbol methodSymbol)
+                {
+                    if (methodSymbol.Name != WellKnownMemberNames.EntryPointMethodName ||
+                        !methodSymbol.IsStatic)
+                    {
+                        return false;
+                    }
+
+                    if (methodSymbol.ReturnsVoid)
+                    {
+                        return true;
+                    }
+
+                    if (methodSymbol.IsAsync &&
```

Comment:
method doesn't have to be async.  it's totally legal to have "public Task Main()".  plz add test.